[NO JIRA]: Updating propTypes package and fixing incorrect proptypes (#2261)
Skyscanner/backpack: UNRELEASED.md

**Fixed:**
- Updated `prop-types` to version `15.7.2`

- bpk-component-barchart:
- bpk-component-calendar:
- bpk-component-datepicker:
- bpk-component-infinite-scroll:
- bpk-theming:

>   - Corrected component props to allow `elementType` for the correct props when components are being used.

Reviewer: ```suggestion
  - Corrected component props to use `elementType` for the props were a component type is expected.
```
